src/components/DecisioningEngine/createEventRegistry.js

```diff
@@ -0,0 +1,98 @@
+/*
+Copyright 2023 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+import { createRestoreStorage, createSaveStorage } from "./utils";
+
+const STORAGE_KEY = "events";
+const MAX_EVENT_RECORDS = 1000;
```

Comment:
Are customers going to hit up against the 5MB sessionStorage limit?

Reply:
We will need to measure the typical size of an event to determine limits, but yes I'd suspect they would eventually run into the limit if we don't cap it . That's why I have introduced `MAX_EVENT_RECORDS`.  For now it's set to 1000, but we can adjust.  I figured limiting to 1000 would keep it well within bounds.